### PR TITLE
Feature/203 vite

### PR DESCRIPTION
**Pull request by patric-eberle**
## Pull request
This PR replaces the webpack builder with Vite. Additionally it improves the TypeScript integration to get better feedback about invalid types. It also extends the test scripts with the TypeScript test. Several packages could be removed because of this change. I also switched the Storybook builder to Vite for better compatibility.
 
### Ticket
https://github.com/valantic/vue-template/issues/203
 
### Browser testing
- *link*
 
### Checklist
- [ ] I merged the current development branch (before testing)
- [ ] Added JSDoc and styleguide demo
- [ ] Tested all links in project relevant browsers
- [ ] Tested all links in different screen sizes
- [ ] Did run automated tests and linters
- [ ] Did assign ticket
- [ ] Double checked target branch
 
## Review/Test checklist
- [ ] Did review code and documentation
- [ ] Tested all links in project relevant browsers
- [ ] Tested all links in different screen sizes
- [ ] Did check accessibility (Wave, only errors)
- [ ] Re-assign ticket to developer

### CODE REVIEW

File: src/components/e-picture.vue [hunk 1]
```diff
@@ -210,40 +204,43 @@
       /**
        * Calculates root element styles.
        */
-      style(): object | null {
+      style(): StyleValue | undefined {
```
Comment (MB-web) [inline src/components/e-picture.vue:207]: What is the reason for changing it to `undefined`?

File: src/pimcore/scss/_gui.scss [hunk 2]
```diff
@@ -0,0 +1,266 @@
+// stylelint-disable selector-class-pattern
```
Comment (MB-web) [inline src/pimcore/scss/_gui.scss:1]: Personally, I would keep pimcore specific files out of the vue boilerplate, but if you want to have it, it's fine.

File: src/setup/i18n.ts [hunk 3]
```diff
@@ -2,6 +2,8 @@ import { createI18n, IntlDateTimeFormat } from 'vue-i18n';
 import fallbackMessages from '../translations/de.json';
 import numberFormats from './localization.json';
 
+type MessagesSchema = typeof fallbackMessages;
```
Comment (MB-web) [inline src/setup/i18n.ts:5]:
```suggestion
type TMessagesSchema = typeof fallbackMessages;
```

File: src/setup/i18n.ts [hunk 4]
```diff
@@ -29,7 +32,7 @@ if (process.env.NODE_ENV !== 'production') {
   }
 }
 
-const i18n = createI18n({
+const i18n = createI18n<[MessagesSchema], 'de'>({
```
Comment (MB-web) [inline src/setup/i18n.ts:35]:
```suggestion
const i18n = createI18n<[TMessagesSchema], 'de'>({
```

File: package.md [hunk 5]
```diff
@@ -48,8 +48,6 @@ This will test the current state of CSS and JS code. Tests **MUST** be executed
   > core
```
Comment (MB-web) [inline package.md:48]: Not urgent, but you probably want to update this file. (I myself can live without this file, as you know ;-) )

File: src/directives/outside-click.ts [hunk 6]
```diff
@@ -21,6 +21,40 @@ interface IOutsideClickElement extends HTMLElement {
   [storageKey]: TOutsideClickHandlerFunction;
 }
 
+function isClickOnExcludedRefElement(excludeRefs: string[], eventTarget: Node, binding: IOutsideClickDirectiveBinding) {
```
Comment (MB-web) [inline src/directives/outside-click.ts:24]: Please add return type

File: src/directives/outside-click.ts [hunk 7]
```diff
@@ -21,6 +21,40 @@ interface IOutsideClickElement extends HTMLElement {
...
+
+    if (excludedElement instanceof HTMLElement) {
+      return excludedElement.contains(eventTarget);
+    }
+
+    return false;
+  });
+}
+
+function isClickOnExcludedIdElement(excludeIds: string[], eventTarget: Node) {
```
Comment (MB-web) [inline src/directives/outside-click.ts:46]: Please add return type

File: src/directives/outside-click.ts [hunk 8]
```diff
@@ -21,6 +21,40 @@ interface IOutsideClickElement extends HTMLElement {
...
+  return excludeIds.some((id) => {
+    const element = document.querySelector(id);
+
+    return element && element.contains(eventTarget);
+  });
+}
+
+function isClickOnExcludedElement(excludeElements: Node[], eventTarget: Node) {
```
Comment (MB-web) [inline src/directives/outside-click.ts:54]: Please add return type

**Review by MB-web — COMMENTED** [review]
Nice job! Seems fast :)

**Comment by patric-eberle** [comment]
@MB-web can you please check my changes? I was also able to remove additional packages 🎉

**Review by MB-web — APPROVED** [review]

**Comment by MB-web** [comment]
@patric-eberle Changes are fine, can be merged from my side.

I have updated `lastDependencyUpdate` in package.json, since you did an Update.

**Comment by patric-eberle** [comment]
> @patric-eberle Changes are fine, can be merged from my side.
> 
> I have updated `lastDependencyUpdate` in package.json, since you did an Update.

@MB-web thank you for the update. Unfortunately we will also have to write a new plugin for vite... else no message will be shown anymore.

**Pull request merged**